module-4/backend/week_four.md

```diff
@@ -5,15 +5,28 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's your favorite project management tool?
+-Waffle.io by far. Has everything I need and is not overly complicated. 
 2. Why do we create staging environments?
+Its important to create staging enviroments so we can keep our production and development serperate and clean. Things on the back end may be different depending on enviroment.
 3. What are the characteristics of a good README (in your opinion)?
+-Gives clear and concise instructions in how to set up the enviroment and run the test suite. It also covers the problem it is solving and brief overview of what it does.
 4. What's one main improvement you're going to make to your code regarding accessibility issues?
+Design with accessibility in mind. Contrasting colors, provide more than just color to convey information, and make sure interactive elements are easy to identify.
 5. What are some basic security concerns to be aware of when building applications?
+Mass assignment, cross-site scripting, and privledge escalation.
 6. Why is continuous integration helpful/important?
+It allows you to keep checks on your application to insure there are no issues while not running development. 
 7. What are some continuous integration tools?
-
+Travis CI, Heroku CI. 
 #### Review  
 
 8. What are some characteristics of "good" git workflow?
+Work off of development, in seperate branches that represent user_stories. Merge branches into development before master.
 9. What are the four fundamental concepts of object oriented programming?
+Encalpsulation:Something defined in an object is hidden from view to the outside. 
+Abstraction: Modeling data to how it looks in real life. 
+Inheritance: A way to resuse code by obtaining elements from another object. 
+Polymorphism: One name many forms. It allows you to use a method in different ways. 
 10. What's a module in Ruby and what's the difference between a class and a module?
+Modules allow you to use methods across multiple classes where methods inside of classes cannot be used in others. The module cannot be instatiated. 
```

Comment:
what are some use cases for using a module vs a class? we could use a method in a class in another class if we inherit from it or we require it and use it